assets/wizards/advertising/views/providers/index.js

```diff
@@ -6,18 +6,71 @@
  * WordPress dependencies
  */
 import { ExternalLink } from '@wordpress/components';
+import { useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
 
 /**
  * Internal dependencies
  */
-import { PluginToggle, ActionCard, withWizardScreen } from '../../../../components/src';
+import {
+	PluginToggle,
+	ActionCard,
+	Modal,
+	Card,
+	Button,
+	withWizardScreen,
+} from '../../../../components/src';
+import GAMOnboarding from '../../components/onboarding';
 
 /**
  * Advertising management screen.
  */
-const Providers = ( { services, toggleService } ) => {
+const Providers = ( { services, fetchAdvertisingData, toggleService } ) => {
 	const { google_ad_manager } = services;
+	const [ inFlight, setInFlight ] = useState( false );
+	const [ networkCode, setNetworkCode ] = useState( '' );
+	const [ isOnboarding, setIsOnboarding ] = useState( false );
+
+	const updateGAMNetworkCode = () => {
+		setInFlight( true );
+		apiFetch( {
+			path: '/newspack/v1/wizard/advertising/network_code/',
+			method: 'POST',
+			data: { network_code: networkCode, is_gam: false },
+			quiet: true,
+		} ).finally( () => {
+			fetchAdvertisingData();
+			setInFlight( false );
+			setIsOnboarding( false );
+		} );
+	};
+
+	let notifications = [];
+
+	if ( google_ad_manager.enabled && google_ad_manager.status.error ) {
+		notifications = notifications.concat( [ google_ad_manager.status.error, '\u00A0' ] );
+	} else if ( google_ad_manager?.created_targeting_keys?.length > 0 ) {
+		notifications = notifications.concat( [
+			__( 'Created custom targeting keys:' ) + '\u00A0',
+			google_ad_manager.created_targeting_keys.join( ', ' ) + '. \u00A0',
+			// eslint-disable-next-line react/jsx-indent
+			<ExternalLink
+				href={ `https://admanager.google.com/${ google_ad_manager.network_code }#inventory/custom_targeting/list` }
+				key="google-ad-manager-custom-targeting-link"
+			>
+				{ __( 'Visit your GAM dashboard' ) }
+			</ExternalLink>,
+		] );
+	}
+
+	if ( google_ad_manager.enabled && ! google_ad_manager.status.connected ) {
+		notifications.push(
+			<Button isLink onClick={ () => setIsOnboarding( true ) }>
```

Comment:
All components inside `notifications` array should have `key` props – otherwise React is unhappy:

```
Warning: Each child in a list should have a unique "key" prop.
```

Reply:
Nice catch! Done in 76314f28e716eed81eeb50853e3df37e8ad97527, which also fixes an issue that was displaying an empty error notice when there were no notifications (`null` should be sent to the component instead of the empty array)